core/pkg/ingress/types.go

```diff
@@ -0,0 +1,236 @@
+	Info() *BackendInfo
+}
+
+// BackendInfo returns information about the backend.
+// This fields contains information that helps to track issues or to
+// map the running ingress controller to source code
+type BackendInfo struct {
+	// Name returns the name of the backend
+	Name string
+	// Release returns the running version (semver)
+	Release string
+	// Build returns information about the git commit
+	Build string
+	// Repository return information about the git repository
+	Repository string
+}
+
+// Configuration holds the definition of all the parts required to describe
+type Configuration struct {
+	// HealthzURL location of the URI that is used to know health state of the ingress controller
+	HealthzURL         string
+	Backends           []*Backend
+	Servers            []*Server
+	TCPEndpoints       []*Location
+	UPDEndpoints       []*Location
+	PassthroughBackend []*SSLPassthroughBackend
+}
+
+// Backend describes one or more remote server/s (endpoints) associated with a service
+type Backend struct {
+	// Name represents an unique api.Service name formatted as <namespace>-<name>-<port>
+	Name string
+	// This indicates if the communication protocol between the backend and the endpoint is HTTP or HTTPS
+	// Allowing the use of HTTPS
+	Secure bool
+	// Endpoints contains the list of endpoints currently running
+	Endpoints []Endpoint
+}
+
+// SSLPassthroughBackend describes a SSL upstream server configured
+// as passthrough (no TLS termination in the ingress controller)
+type SSLPassthroughBackend struct {
+	Backend
+
+	Host string
+}
+
+// BackendByNameServers sorts upstreams by name
+type BackendByNameServers []*Backend
+
+func (c BackendByNameServers) Len() int      { return len(c) }
+func (c BackendByNameServers) Swap(i, j int) { c[i], c[j] = c[j], c[i] }
+func (c BackendByNameServers) Less(i, j int) bool {
+	return c[i].Name < c[j].Name
+}
+
+// Endpoint describes a kubernetes endpoint in an backend
+type Endpoint struct {
+	// Address IP address of the endpoint
+	Address string
+	// Port number of the TCP port
+	Port string
+	// MaxFails returns the maximum number of check failures
+	// allowed before this should be considered dow.
+	// Setting 0 indicates that the check is performed by a Kubernetes probe
+	MaxFails int
+
+	FailTimeout int
+}
+
+// Server describes a virtual server
+type Server struct {
+	// Name contains the FQDN
+	Name string
+	// SSL indicates if the virtual server is configured for TLS
+	SSL bool
+	// SSLPassthrough indicates if the TLS termination is realized in the server or in the remote endpoint
+	SSLPassthrough bool
+
+	SSLCertificate string
+	//SSLCertificateKey string
+	SSLPemChecksum string
+	// Locations list of URIs configured in the server
+	Locations []*Location
+}
+
+// Location describes an URI inside a virtual server.
+type Location struct {
+	IsDefBackend    bool
+	SecureBackend   bool
+	EnableCORS      bool
+	Path            string
+	Backend         Backend
```

Comment:
it's confusing that we have a list of backensd in `Configuration` as well as a Backend in here.